addons/longhorn/template/base/install.sh

```diff
@@ -119,3 +122,30 @@ function longhorn_preflight() {
     local src="${DIR}/addons/longhorn/${LONGHORN_VERSION}"
     echo "${src}/host-preflight.yaml"
 }
+
+function check_mount_propagation() {
+    local src=$1
+
+    kubectl get ns longhorn-system 2>/dev/null 1>/dev/null || kubectl create ns longhorn-system 1>/dev/null
+
+    render_yaml_file "$src/tmpl-mount-propagation.yaml" > "$src/mount-propagation.yaml"
+    kubectl create -f "$src/mount-propagation.yaml"
+    echo "Waiting for the Longhorn Mount Propagation Check Daemonset to be ready"
+    spinner_until 120 longhorn_daemonset_is_ready longhorn-manager
+
+    validate_longhorn_ds
+
+    kubectl delete -f "$src/mount-propagation.yaml"
+}
+
+
+validate_longhorn_ds() {
+    local allSupported=true
+
+    local allpods=$(kubectl get daemonsets -n longhorn-system longhorn-environment-check --no-headers | tr -s ' ' | cut -d ' ' -f4)
```

Comment:
if this fails allpods will be empty. can you account for that in your conditional on line 148?